### PR TITLE
New generic set_services is now in place with the legacy interface being

### PR DESCRIPTION
**Pull request by tclune**
just a deprecated stub.

## Description

Implemented a new generic setservices that is designed to wrap user setservices.  This is as opposed to the legacy generic setservices that is called from within user setservices.  That legacy procedure is now an empty (deprecated) subroutine.

## Motivation and Context
Small step to supporting NUOPC.

## How Has This Been Tested?

Tested with C12 on laptop with ifort.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist:

- [X] I have tested this change with a run of GEOSgcm (if non-trivial)
- [X] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [X] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

File: gridcomps/History/MAPL_HistoryGridComp.F90
Comment (mathomp4) [inline gridcomps/History/MAPL_HistoryGridComp.F90:926]: Is this a print on every process?

**Comment by mathomp4** [comment]
I think we are hitting a build issue in the GEOSldas where they need an update (https://github.com/GEOS-ESM/GEOSldas/pull/515) to build GEOSldas. A PR was accepted in GEOSgcm_GridComp and they couldn't update the LDAS until that PR was merged.

Now that it is, they have a new PR to let it all work again.

**Comment by gmao-rreichle** [comment]
> I think we are hitting a build issue in the GEOSldas where they need an update ([GEOS-ESM/GEOSldas#515](https://github.com/GEOS-ESM/GEOSldas/pull/515)) to build GEOSldas. A PR was accepted in GEOSgcm_GridComp and they couldn't update the LDAS until that PR was merged.
> 
> Now that it is, they have a new PR to let it all work again.

@mathomp4:  As far as I can tell, the "build issue in GEOSldas" was because I renamed a constant in the GCM GC that is also used in GEOSldas.  Most times when there are land changes in GCM GC (typically by us), we need a matching GEOSldas PR to make it all work.  I don't think this has to do with generic or legacy MAPL interfaces.  Maybe I'm missing something...

**Comment by mathomp4** [comment]
> > I think we are hitting a build issue in the GEOSldas where they need an update ([GEOS-ESM/GEOSldas#515](https://github.com/GEOS-ESM/GEOSldas/pull/515)) to build GEOSldas. A PR was accepted in GEOSgcm_GridComp and they couldn't update the LDAS until that PR was merged.
> > Now that it is, they have a new PR to let it all work again.
> 
> @mathomp4: As far as I can tell, the "build issue in GEOSldas" was because I renamed a constant in the GCM GC that is also used in GEOSldas. Most times when there are land changes in GCM GC (typically by us), we need a matching GEOSldas PR to make it all work. I don't think this has to do with generic or legacy MAPL interfaces. Maybe I'm missing something...
> 
> **Update:** Actually, it wasn't the renaming of a constant in this case, but rather the intentional change in the interface of subroutine catch_calc_soil_moist() in the GCM GC. But same message - this was caused (intentional) land changes.

@gmao-rreichle Yes. I just wanted to let @tclune know that the issue wasn't in his PR. It's just part of all the changes GEOS has been having today due to fun repo updates. The GEOSgcm was also broken in CI for a while today until Scott and I could get things synced up. (And, weirdly, the Chem repo *didn't* break but that was due to a bug in my CI. Fun day for CI!)

**Comment by mathomp4** [comment]
Re-submitted the CI. Let's do this! 🚀

**Review by mathomp4 — DISMISSED** [review]
All passed for me. I'll let @atrayano take a look as I have no idea what most of this does 😄

File: gridcomps/Cap/MAPL_CapGridComp.F90
Comment (atrayano) [inline gridcomps/Cap/MAPL_CapGridComp.F90:149]: I cannot see where we use the variable 'root_name', besides getting it from a resource file

Comment (tclune) [reply gridcomps/Cap/MAPL_CapGridComp.F90:149]: OK - I'll take a look. Presumably it is used to give the name of the ROOT GC.

Comment (tclune) [reply gridcomps/Cap/MAPL_CapGridComp.F90:149]: Must have been something "lost" when moving chunks of code around.  Definitely not needed here - as it is doing nothing.
  I will fix.

Comment (atrayano) [reply gridcomps/Cap/MAPL_CapGridComp.F90:149]: This is what I thought too, but I did not see it (plus the variable scope is limited to the block)

**Review by atrayano — DISMISSED** [review]

**Comment by mathomp4** [comment]
Grah! What is happening with the adas test?

**Comment by mathomp4** [comment]
This is zero-diff! But blocking until @tclune says it's good to go (so I don't accidentally merge in)

**Review by mathomp4 — APPROVED** [review]
This does work and is zero-diff

**Pull request merged**
